enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/ClientEvaluationService.java

```diff
@@ -103,33 +98,49 @@ public void initClientEvaluation(final OwnerId ownerId, final IdentityVerificati
     public void processClientEvaluation(final IdentityVerificationEntity identityVerification, final OwnerId ownerId) {
         logger.debug("Client evaluation started for {}", identityVerification);
 
+        final String verificationId;
+        try {
+            verificationId = getVerificationId(identityVerification);
+        } catch (Exception e) {
+            processVerificationIdError(identityVerification, ownerId, e);
+            return;
+        }
+
         final EvaluateClientRequest request = EvaluateClientRequest.builder()
                 .processId(identityVerification.getProcessId())
                 .userId(identityVerification.getUserId())
                 .identityVerificationId(identityVerification.getId())
-                .verificationId(getVerificationId(identityVerification))
+                .verificationId(verificationId)
                 .build();
 
         final int maxFailedAttempts = config.getClientEvaluationMaxFailedAttempts();
         for (int i = 0; i < maxFailedAttempts; i++) {
             try {
                 final EvaluateClientResponse response = onboardingProvider.evaluateClient(request);
                 processEvaluationSuccess(identityVerification, ownerId, response);
-                break;
-            } catch (Throwable t) {
-                processEvaluationError(identityVerification, ownerId, t);
+                logger.debug("Client evaluation finished for {}, attempt: {}", identityVerification, i + 1);
+                return;
+            } catch (Exception e) {
+                logger.warn("Client evaluation failed for {}, attempt: {}, {}, {}", identityVerification, i + 1, ownerId, e.getMessage());
+                logger.debug("Client evaluation failed for {} - attempt: {}, {}", identityVerification, i + 1, ownerId, e);
             }
         }
-        logger.debug("Client evaluation finished for {}", identityVerification);
+        processTooManyEvaluationError(identityVerification, ownerId);
     }
 
     private static String getVerificationId(final IdentityVerificationEntity identityVerification) {
-        return identityVerification.getDocumentVerifications().stream()
+        final Set<String> verificationIds = identityVerification.getDocumentVerifications().stream()
                 .filter(DocumentVerificationEntity::isUsedForVerification)
                 .filter(it -> it.getStatus() == DocumentStatus.ACCEPTED)
-                .findAny()
                 .map(DocumentVerificationEntity::getVerificationId)
-                .orElseThrow(() -> new IllegalStateException("No accepted document verification for " + identityVerification));
+                .collect(toSet());
+
+        if (verificationIds.size() == 1) {
```

Comment:
The main goal of the issue was about this change, but it turned out that the service deserves larger cleanup.